contrib/pyln-client/pyln/client/gossmap.py

```diff
@@ -50,9 +51,25 @@ def __init__(self, channel: 'GossmapChannel', direction: int,
         self.fee_base_msat: int = fee_base_msat
         self.fee_proportional_millionths: int = fee_proportional_millionths
 
+        # Cache the _scidd and hash to have faster operation later
+        # Unfortunately the @final decorator only comes for python3.8
+        self._scidd = f"{self.channel.scid}x{self.direction}"
+        self._hash = self._scidd.__hash__()
```

Comment:
by "u64 representation" you mean accessing the parents `self.channel`  blocknum, txnum, outnum and combine them somehow with the direction into a u64?

Reply:
I think we can probably just keep a `numscid` in the `ShortChannelId` and use that. Depending on access pattern it might be sensible to switch the internal representation around too (store just the u64 value, rather than destructuring it). This is true especially if we just care about scid equality and lookups, and we're not as interested in the blockheight for example.

Reply:
And by numeric representation I mean:

```python
numscid = block << 40 | txindex << 16 | outnum
```

And the one with direction could then be:

```python
numscidd = direction << 63 | numscid
```

Reply:
Yes why not, that's a bit faster.
I also added a `__hash__()` to the `GossmapChannel` that simply uses the hash of the channels `ShortChannelId` (which uses your numscid internally already).
That would be helpful if someone operates on Sets and Maps of  `GossmapChannel`